File: sdks/go/sdks/go/sdks/python/publish.py

#!/usr/bin/env python3
import subprocess
import os
import sys

def run_command(command):
    print(f"Running: {command}")
    result = subprocess.run(command, shell=True)
    if result.returncode != 0:
        print(f"Command failed with exit code {result.returncode}")
        sys.exit(result.returncode)

# Clean previous builds
if os.path.exists("dist"):
    run_command("rm -rf dist")
if os.path.exists("build"):
    run_command("rm -rf build")
if os.path.exists("udene_sdk.egg-info"):
    run_command("rm -rf udene_sdk.egg-info")

# Build the package
run_command("python setup.py sdist bdist_wheel")

# Upload to PyPI
run_command("twine upload dist/*")

print("Package published successfully!")

File: sdks/go/sdks/ruby/publish.rb

#!/usr/bin/env ruby
# frozen_string_literal: true

require "fileutils"

def run_command(command)
  puts "Running: #{command}"
  system(command)
  exit($?.exitstatus) unless $?.success?
end

# Clean previous builds
FileUtils.rm_rf("pkg") if Dir.exist?("pkg")

# Build the gem
run_command("gem build udene_sdk.gemspec")

# Get the built gem file
gem_file = Dir.glob("udene_sdk-*.gem").first
raise "No gem file found" unless gem_file

# Push to RubyGems
run_command("gem push #{gem_file}")

puts "Gem published successfully!"
